### ISSUE TITLE
Process Applications UI

### ISSUE THREAD

**Issue by philippfromme** [camunda/camunda-modeler#4664]
### Problem you would like to solve

For the new [process applications feature](https://github.com/camunda/camunda-modeler/issues/2508) a UI will be required. Users should be able to

* [ ] Create a new process application
* [ ] Open an existing process application
* [ ] Deploy a process application
* [ ] Link resrouces of a process application (cf. https://docs.camunda.io/docs/components/modeler/web-modeler/advanced-modeling/form-linking/)

### Proposed solution

There are a few options for creating a UI based on existing UI patterns in the Desktop Modeler:

### 1. New bottom panel tab

![image](https://github.com/user-attachments/assets/de9d9264-1fe8-4c52-9f47-773825d47bc0)

### 2. New side panel

![image](https://github.com/user-attachments/assets/999e13f6-7472-4a0e-9745-6f2fd359f66c)
 
### 3. New modal

![image](https://github.com/user-attachments/assets/e1031dcf-8def-4675-977a-24d0b9241c32)

### 4. New tab

![image](https://github.com/user-attachments/assets/b350b156-98af-4db3-9e8a-907c80ac2cd3)

### 5. New overlay

![image](https://github.com/user-attachments/assets/36d0a46e-d91d-46a1-9fdc-a37db606c71b)

### Alternatives considered

There is no alternative that requires no UI at all.

### Additional context

Related to https://github.com/camunda/camunda-modeler/issues/2508

**Comment by nikku** [comment]
CC @lmbateman.

**Comment by philippfromme** [comment]
We discussed possible low-UI but high value features with @barmac. Linking resources could work in two simple ways:

1. Autocompletion in the ID input

![image](https://github.com/user-attachments/assets/abeaa6a6-e7df-46a8-a30b-2d58723bbcbf)

2. Creating elements with linked resources (e.g. create call activity with process linked already)

![image](https://github.com/user-attachments/assets/d672fc4b-2b03-44e7-859f-a0def232ed6b)

**Comment by nikku** [comment]
Love both options, and goes in line with doing more of the hard work (to make it simple).

The later one would probably still refer to call activity (link to payment process).

**Comment by philippfromme** [comment]
> The later one would probably still refer to call activity (link to payment process)? We want to ensure we don't do the connectors thing and hide the BPMN purpose of the element to be appended.

Yes, I just quickly hacked this together in the browser. 🙂

**Comment by barmac** [comment]
> Love both options, and goes in line with doing more of the hard work (to make it simple).
> 
> The later one would probably still refer to call activity (link to payment process)? We want to ensure we don't do the connectors thing and hide the BPMN purpose of the element to be appended.

Yes, this would append a call activity with the processId (link) and name preset. The append menu would use the call activity icon for full transparency.

**Comment by nikku** [comment]
For the moment I think we're fine with the low-UI approach. Let's close this issue, and await user feedback?

**Comment by barmac** [comment]
We are actually implementing the UI via https://github.com/camunda/camunda-modeler/pull/4762, so this issue will be closed when the PR is merged.

**Comment by barmac** [comment]
Indeed it is low UI, but still UI ^^

**Comment by philippfromme** [comment]
Closed by https://github.com/camunda/camunda-modeler/pull/4762

**Comment by Buckwich** [comment]
@philippfromme this was already released with M87, right? Or did you add major features after the release?

**Comment by philippfromme** [comment]
I consider this to be released with 5.34 already, yes.
